storage/transfermanager/downloader.go

```diff
@@ -41,51 +41,45 @@ type Downloader struct {
 }
 
 // DownloadObject queues the download of a single object. This will initiate the
-// download but is non-blocking; call Downloader.Results to process the result.
-func (d *Downloader) DownloadObject(ctx context.Context, input *DownloadObjectInput) {
-	input.ctx = ctx
-	d.addInput(input)
-}
+// download but is non-blocking; call Downloader.Results or use the callback to
+// process the result.
```

Comment:
We should explicitly note that this is intended to be thread-safe (can share the Downloader across goroutines). And add the note about timeouts/timing that you have in the example.

Reply:
Done. Note I also made WaitAndClose callable more than once (just returns the same instead of panicking).